pkg/kv/kvserver/replica_app_batch.go

```diff
@@ -410,12 +415,23 @@ func (b *replicaAppBatch) runPostAddTriggersReplicaOnly(
 		// going to bother with a long-running migration.
 		apply := !looselyCoupledTruncation || res.RaftExpectedFirstIndex == 0
 		if apply {
+			// TODO(pavelkalinnikov): counting bytes may be unreliable because 0 bytes
+			// does not mean 0 files. Make the interface more robust.
+			if rmBytes, _, err := b.r.raftMu.sideloaded.BytesIfTruncatedFromTo(
+				ctx, 0, res.State.TruncatedState.Index,
+			); err != nil {
+				return errors.Wrap(err, "unable to count sideloaded files to remove")
+			} else if rmBytes != 0 {
+				b.changeTruncatesSideloadedFiles = true
+			}
 			if apply, err = handleTruncatedStateBelowRaftPreApply(
 				ctx, b.state.TruncatedState, res.State.TruncatedState, b.r.raftMu.stateLoader, b.batch,
 			); err != nil {
 				return errors.Wrap(err, "unable to handle truncated state")
 			}
 		} else {
+			// TODO(pavelkalinnikov): ensure that this truncation path makes sure the
```

Comment:
@sumeerbhola Do you know if it's true already? Is the decoupled truncation also susceptible to the same problem that this PR fixes?